distributed/comm/ucx.py

```diff
@@ -596,14 +612,27 @@ def _scrub_ucx_config():
         if dask.config.get("distributed.comm.ucx.nvlink"):
             tls = tls + ",cuda_ipc"
 
-        options = {"TLS": tls, "SOCKADDR_TLS_PRIORITY": tls_priority}
+        high_level_options = {"TLS": tls, "SOCKADDR_TLS_PRIORITY": tls_priority}
+
+    # Pick up any other ucx environment settings
+    # {"some-name": value} is translated to {"UCX_SOME_NAME": value}
+    def normalise(k):
+        return "_".join(map(str.upper, ("UCX", *k.split("-"))))
+
+    environment_options = {
+        key: v
+        for k, v in dask.config.get("distributed.comm.ucx.environment", {}).items()
+        # High-level settings take precedence over low-level
+        # environment configuration, as do settings already in the
+        # environment (set externally)
+        if (key := normalise(k)) not in os.environ and key[4:] not in high_level_options
+    }
 
-    # ANY UCX options defined in config will overwrite high level dask.ucx flags
-    valid_ucx_vars = list(get_config().keys())
-    for k, v in options.items():
+    valid_ucx_vars = set(get_config().keys())
+    for k, v in high_level_options.items():
         if k not in valid_ucx_vars:
             logger.debug(
                 f"Key: {k} with value: {v} not a valid UCX configuration option"
             )
```

Comment:
I think we will need to remove this. Unfortunately we can't validate non-UCP options, so if someone tries to set `distributed.comm.ucx.environment.tcp-rx-seg-size` for example, this is likely to throw messages that are not correct. Alternatively, we could keep this and make it more explicit, e.g., `... not a valid UCP configuration ...`, but that may be too confusing for those that do not know much about UCX layers.

Reply:
Note that this is only validating `high_level_options` (which can only contain `TLS` and `TLS_SOCKADDR_PRIORITY` by construction). So we won't have anything non-UCP here. We could just remove this because the only way slots get set in `high_level_options` is by this function anyway.

I think this is a leftover from when one could put arbitrary key-value pairs in the ucx config dict (about 2 years ago).

Reply:
Yeah, I actually overlooked the fact this is validating `high_level_options`, but I agree in that case it doesn't make any sense to keep it, so I would just remove this.

Reply:
Done.